packages/core-data/src/actions.js

```diff
@@ -203,36 +205,29 @@ export function* deleteEntityRecord(
 				path = addQueryArgs( path, query );
 			}
 
-			const options = {
+			deletedRecord = await __unstableFetch( {
 				path,
 				method: 'DELETE',
-			};
-			if ( __unstableFetch ) {
-				deletedRecord = yield __unstableAwaitPromise(
-					__unstableFetch( options )
-				);
-			} else {
-				deletedRecord = yield apiFetch( options );
-			}
+			} );
 
-			yield removeItems( kind, name, recordId, true );
+			await dispatch( removeItems( kind, name, recordId, true ) );
```

Comment:
Sync action that removes items from the in-memory cache in the store.

Reply:
I am still not on board with removing `await` when we dispatch a function call because it assumes we know the implementation details of that function. I think we're pretty close to having all the actions refactored though so maybe that would be enough to start the larger discussion? I have the resolvers in mind too, but maybe they're not a blocker for that.

Reply:
> it assumes we know the implementation details of that function.

We only need to know the type of what the `removeItems` function returns. If that's a function (thunk) that returns a promise, we might want to await it. We already know what the parameters of `removeItems` are, and it's not an implementation details either.